src/hooks/index.js

```diff
@@ -234,29 +234,49 @@ exports.mapPaginationQuery = (hook) => {
 	}
 };
 
-exports.checkCorrectCourseId = (hook) => {
-	let courseService = hook.app.service('courses');
-	const courseId = (hook.data.courseId || '').toString() || (hook.id || '').toString();
-	let query = {};
+exports.checkCorrectCourseOrTeamId = async (hook) => {
 
-	if (hook.data.courseGroupId) {
-		delete hook.data.courseId;
-		query = {$or: [{teacherIds: {$in: [hook.params.account.userId]}}, {userIds: {$in: [hook.params.account.userId]}}]};
-	} else
-		query = { teacherIds: {$in: [hook.params.account.userId] } };
+	if (hook.data.teamId) {
+		let teamService = hook.app.service('teams');
 
-	return courseService.find({ query: query})
-		.then(courses => {
-			if (courses.data.some(course => { return course._id.toString() === courseId; }))
-				return hook;
-			else
-				throw new errors.Forbidden("The entered course doesn't belong to you!");
-		});
+		let query = {
+			userIds: {
+				$elemMatch: { userId: hook.params.account.userId }
+			}
+		};
+
+		let teams = await teamService.find({ query });
+
+		if (teams.data.some(team => team._id.toString() === hook.data.teamId )) {
+			return hook;
+		} else {
+			throw new errors.Forbidden("The entered team doesn't belong to you!");
+		}
+	} else if (hook.data.courseGroupId || hook.data.courseId) {
+		let courseService = hook.app.service('courses');
+		const courseId = (hook.data.courseId || '').toString() || (hook.id || '').toString();
+		let query = { teacherIds: {$in: [hook.params.account.userId] } };
```

Comment:
Please correct me if I'm wrong, but is `$in` used correctly here? Why use `$in` if only one value is supplied?

https://github.com/feathersjs/docs/blob/master/api/databases/querying.md#in-nin